services/datamanager/builtin/builtin.go

```diff
@@ -730,6 +737,21 @@ func pollFilesystem(ctx context.Context, wg *sync.WaitGroup, captureDir string,
 		case <-ctx.Done():
 			return
 		case <-t.C:
+			logger.Debug("Checking disk usage")
+			shouldDelete, err := shouldDeleteBasedOnDiskUsage(ctx, captureDir, logger)
+			if err != nil {
+				logger.Errorw("Error checking file system stats", "error", err)
+			}
+			if shouldDelete {
+				start := time.Now()
+				deletedFileCount, err := deleteFiles(ctx, syncer, captureDir, logger)
+				duration := time.Since(start)
+				if err != nil {
+					logger.Errorw("Error deleting cached datacapture files", "error", err, "execution time", duration.Seconds())
+				} else {
+					logger.Infof("%v files have been deleted to avoid the disk filling up, execution time: %f", deletedFileCount, duration.Seconds())
```

Comment:
I'm mostly giving comments around logs since it's super helpful for the user debugging what's going on, especially when looking back at logs while the machine is offline for a while — worth also adding "Deleted every %d" file, deleteEveryNth

Reply:
Talked offline, handled in PR for making N configurable.